### PR TITLE
Matched func_802A1628_795908

### PR DESCRIPTION
**Pull request by Nasina7**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.00%) on `us`.

File: src/battle/star/up_and_away/7952E0.c [hunk 1]
```diff
@@ -43,23 +43,28 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
...
-    Actor* targetActor = get_actor(get_actor(script->owner1.actorID)->targetActorID);
-    s32 flags = targetActor->flags;
-
+    Actor* targetActor = get_actor((s32)get_actor(script->owner1.actorID)->targetActorID);
+    
     script->varTable[0] = 0;
 
-    if ((flags & ACTOR_FLAG_TARGET_ONLY) == NULL && (flags & ACTOR_FLAG_NO_DMG_APPLY) == NULL && (flags & ACTOR_FLAG_2000) == NULL && targetActor->actorBlueprint->upAndAwayChance != 0 && targetActor->actorBlueprint->upAndAwayChance >= rand_int(100)) {
-        script->varTable[0] = 1;
+    {
```
Comment (pixel-stuck) [inline src/battle/star/up_and_away/7952E0.c:51]: Is there a reason for the extra scoping?

File: src/battle/star/up_and_away/7952E0.c [hunk 2]
```diff
@@ -43,23 +43,28 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-#ifdef NON_EQUIVALENT
 ApiStatus func_802A1628_795908(Evt* script, s32 isInitialCall) {
-    Actor* targetActor = get_actor(get_actor(script->owner1.actorID)->targetActorID);
-    s32 flags = targetActor->flags;
-
+    Actor* targetActor = get_actor((s32)get_actor(script->owner1.actorID)->targetActorID);
```
Comment (pixel-stuck) [inline src/battle/star/up_and_away/7952E0.c:47]: s32 cast here should be eliminated if possible

File: src/battle/star/up_and_away/7952E0.c [hunk 3]
```diff
@@ -43,23 +43,28 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
...
+    
     script->varTable[0] = 0;
 
-    if ((flags & ACTOR_FLAG_TARGET_ONLY) == NULL && (flags & ACTOR_FLAG_NO_DMG_APPLY) == NULL && (flags & ACTOR_FLAG_2000) == NULL && targetActor->actorBlueprint->upAndAwayChance != 0 && targetActor->actorBlueprint->upAndAwayChance >= rand_int(100)) {
-        script->varTable[0] = 1;
+    {
+        u32 flags = targetActor->flags;
+
+        if((flags & ACTOR_FLAG_TARGET_ONLY) == 0) {
```
Comment (pixel-stuck) [inline src/battle/star/up_and_away/7952E0.c:54]: Flag checks should just be done as
`if (!(flags & mask))` goes for pretty much everything here

File: src/battle/star/up_and_away/7952E0.c (hunk 1 above)
Comment (Nasina7) [reply src/battle/star/up_and_away/7952E0.c:51]: The script->varTable[0] = 0; line needs to come before the flag declaration, but declaring the flags variable afterwards gives a parsing error.  Putting it in another scope fixes the error.  I'm more used to C++ than C so I don't know of any other way to fix this, but if you know how to, then let me know.

Comment (pixel-stuck) [reply src/battle/star/up_and_away/7952E0.c:51]: Yeah, this is a c89 oddity (all variables need to be declared at the beginning of a scope). Are you sure you can't define the var with no assignment and then assign it after the script->varTable[0] assignment?

Comment (Nasina7) [reply src/battle/star/up_and_away/7952E0.c:51]: Oh, yeah that worked.  I've no idea why I didn't try that lol.

File: src/battle/star/up_and_away/7952E0.c [hunk 4]
```diff
@@ -43,23 +43,26 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
...
-    if ((flags & ACTOR_FLAG_TARGET_ONLY) == NULL && (flags & ACTOR_FLAG_NO_DMG_APPLY) == NULL && (flags & ACTOR_FLAG_2000) == NULL && targetActor->actorBlueprint->upAndAwayChance != 0 && targetActor->actorBlueprint->upAndAwayChance >= rand_int(100)) {
-        script->varTable[0] = 1;
+    flags = targetActor->flags;
+    
+    if(!(flags & ACTOR_FLAG_TARGET_ONLY)) {
+        if(!(flags & ACTOR_FLAG_NO_DMG_APPLY)) {
+            if (!(flags & ACTOR_FLAG_2000) && targetActor->actorBlueprint->upAndAwayChance != 0 && rand_int(100) <= targetActor->actorBlueprint->upAndAwayChance) {
+                script->varTable[0] = 1;
+            }
+        }
```
Comment (ethteck) [inline src/battle/star/up_and_away/7952E0.c:58]:
```suggestion
        } else {
```

File: src/battle/star/up_and_away/7952E0.c [hunk 5]
```diff
@@ -43,23 +43,26 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-#ifdef NON_EQUIVALENT
 ApiStatus func_802A1628_795908(Evt* script, s32 isInitialCall) {
     Actor* targetActor = get_actor(get_actor(script->owner1.actorID)->targetActorID);
-    s32 flags = targetActor->flags;
-
+    u32 flags;
```
Comment (ethteck) [inline src/battle/star/up_and_away/7952E0.c:48]: u32 flags = ...

**Review by ethteck — CHANGES_REQUESTED** [review]
Nice! Just minor code style things

File: src/battle/star/up_and_away/7952E0.c (hunk 5 above)
Comment (pixel-stuck) [reply src/battle/star/up_and_away/7952E0.c:48]: Per previous conversation, assignment has to be in this order

**Review by ethteck — APPROVED** [review]
lgtm! just need the alex review and we can merge

File: src/battle/star/up_and_away/7952E0.c [hunk 6]
```diff
@@ -43,23 +43,25 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
...
+    
     script->varTable[0] = 0;
-
-    if ((flags & ACTOR_FLAG_TARGET_ONLY) == NULL && (flags & ACTOR_FLAG_NO_DMG_APPLY) == NULL && (flags & ACTOR_FLAG_2000) == NULL && targetActor->actorBlueprint->upAndAwayChance != 0 && targetActor->actorBlueprint->upAndAwayChance >= rand_int(100)) {
-        script->varTable[0] = 1;
+    flags = targetActor->flags;
+    
+    if(!(flags & ACTOR_FLAG_TARGET_ONLY)) {
+        if(!(flags & ACTOR_FLAG_NO_DMG_APPLY)) {
```
Comment (nanaian) [inline src/battle/star/up_and_away/7952E0.c:54]:
```suggestion
    if (!(flags & ACTOR_FLAG_TARGET_ONLY)) {
        if (!(flags & ACTOR_FLAG_NO_DMG_APPLY)) {
```

**Review by nanaian — CHANGES_REQUESTED** [review]
Nice match!

File: src/battle/star/up_and_away/7952E0.c (hunk 6 above)
Comment (Nasina7) [reply src/battle/star/up_and_away/7952E0.c]: :+1:

**Review by nanaian — APPROVED** [review]

**Pull request merged**